lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileSystem.java

```diff
@@ -302,9 +302,12 @@ public void createDirectory(Location location)
                     throw new IOException("mkdirs failed");
                 }
                 // explicitly set permission since the default umask overrides it on creation
-                if (permission.isPresent()) {
-                    fileSystem.setPermission(directory, permission.get());
-                }
+                // CCCS-MODIFICATION: Folders are already created with the proper permissions and this
```

Comment:
```suggestion
                // CCCS-MODIFICATION: Folders are already created with the proper permissions and
```
Also, permissions can't be what? Changed?

Reply:
oups! Yes permissions can`t be changed.